### PR TITLE
QUAD-110 : Created "Project Blog" navigation from public page.

### PR DESCRIPTION
**Pull request by kavinya**
QUAD-110 : Created "Project Blog" navigation from public page. Changes are made in navbar.jsp, tiles-def.xml, WebsiteProjectController. Added a new view as "projectblog.jsp".

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java [hunk 1]
```diff
@@ -202,4 +202,24 @@ public String visualizeAllNetworks(@PathVariable("projectUnixName") String proje
...
+
+	/**
+	 * This method gives the the projectblog
+	 * @param projectUnixName	The project unix name
+	 * @param model				Model
+	 * @return view
+	 * @throws QuadrigaStorageException
+	 */
+	@RequestMapping(value = "sites/{projectUnixName}/projectblog", method = RequestMethod.GET)
+	public String projectblog(@PathVariable("projectUnixName") String projectUnixName,
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java]: this method should be in its own class

File: Quadriga/src/main/webapp/resources/txt-layout/js/d3networks.js [hunk 2]
```diff
@@ -634,4 +634,5 @@ function d3visualize(graph, networkId, path,type) {
 
 	}
 
-}
\ No newline at end of file
+}
+
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/resources/txt-layout/js/d3networks.js]: Do those changes belong to the blog story as well?

File: Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp [hunk 3]
```diff
@@ -1,4 +1,5 @@
 <li ${currentPage == "home" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
 <li ${currentPage == "networks" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
 <li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
-<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
\ No newline at end of file
+<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
+<li ${currentPage == "projectblog" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/projectblog">Project Blog</a></li>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp]: This link should be right next to the "Home" link in the navigation bar.

**Comment by kavinya** [comment]
@jdamerow Resolved merge conflicts. Please verify

**Comment by jdamerow** [comment]
@kavinya There are still merge conflicts. please resolve them.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+public class PublicBlogController {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java:22]: javadoc missing

File: Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp [hunk 5]
```diff
@@ -1,10 +1,5 @@
...
-<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/search">Search</a></li>
-<li ${currentPage == "getprojectstatistics" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/statistics">Statistics</a></li>
\ No newline at end of file
+<li ${currentPage == "home" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
+<li ${currentPage == "projectblog" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/projectblog">Project Blog</a></li>
+<li ${currentPage == "networks" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
+<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp]: You're still deleting the links from the other version. Please fix that.

**Comment by kavinya** [comment]
@jdamerow  Resolved conflicts for this issue.

File: Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp [hunk 6]
```diff
@@ -1,10 +1,13 @@
 <li ${currentPage == "home" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
+	href="${pageContext.servletContext.contextPath}/sites/${project.unixName}">Home</a></li>
 <li ${currentPage == "networks" ? "class=\"active\"" : ""}><a
-        href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/browsenetworks">Networks</a></li>
+	href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/projectblog">Project
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp:4]: you probably want to adjust the currentPage value here. the current page is not "networks", is it?

**Pull request closed without merging**